airbyte_cdk/sources/declarative/declarative_component_schema.yaml

```diff
@@ -3002,79 +2984,38 @@ definitions:
         description: Component decoding the response so records can be extracted.
         anyOf:
           - "$ref": "#/definitions/CustomDecoder"
+          - "$ref": "#/definitions/CsvDecoder"
           - "$ref": "#/definitions/JsonDecoder"
           - "$ref": "#/definitions/JsonlDecoder"
           - "$ref": "#/definitions/IterableDecoder"
           - "$ref": "#/definitions/XmlDecoder"
-          - "$ref": "#/definitions/GzipJsonDecoder"
-          - "$ref": "#/definitions/CompositeRawDecoder"
           - "$ref": "#/definitions/ZipfileDecoder"
       $parameters:
         type: object
         additionalProperties: true
-  CompositeRawDecoder:
-    description: "(This is experimental, use at your own risk)"
+  GzipDecoder:
     type: object
     required:
       - type
-      - parser
+      - inner_decoder
     properties:
       type:
         type: string
-        enum: [CompositeRawDecoder]
-      parser:
+        enum: [GzipDecoder]
+      inner_decoder:
         anyOf:
-          - "$ref": "#/definitions/GzipParser"
-          - "$ref": "#/definitions/JsonParser"
-          - "$ref": "#/definitions/JsonLineParser"
-          - "$ref": "#/definitions/CsvParser"
-  #    PARSERS
-  GzipParser:
-    type: object
-    required:
-      - type
-      - inner_parser
-    properties:
-      type:
-        type: string
-        enum: [GzipParser]
-      inner_parser:
-        anyOf:
-          - "$ref": "#/definitions/JsonLineParser"
-          - "$ref": "#/definitions/CsvParser"
-          - "$ref": "#/definitions/JsonParser"
-  JsonParser:
-    title: JsonParser
-    description: Parser used for parsing str, bytes, or bytearray data and returning data in a dictionary format.
-    type: object
-    required:
-      - type
-    properties:
-      type:
-        type: string
-        enum: [JsonParser]
-      encoding:
-        type: string
-        default: utf-8
-  JsonLineParser:
-    type: object
-    required:
-      - type
-    properties:
-      type:
-        type: string
-        enum: [JsonLineParser]
-      encoding:
-        type: string
-        default: utf-8
-  CsvParser:
+          - "$ref": "#/definitions/CsvDecoder"
```

Comment:
Nothing stops us from shoving XmlDecoder here as well, but we're just not sure if there is a usecase for it?

Reply:
Those four entries have an associated parser. [XmlDecoder](https://github.com/airbytehq/airbyte-python-cdk/blob/74631d8850510194d90faeddbb489ada5ad3cb8a/airbyte_cdk/sources/declarative/decoders/xml_decoder.py#L19) does not as it was implemented as a Decoder and not a Parser. Hence, it can't be supported in underlying decoders like in GzipDecoder